### DISCUSSION TITLE
Fluidsynth master  fine tune

### DISCUSSION THREAD

**Discussion by radionut44** [FluidSynth/fluidsynth#807] [Q&A] [answered]
In a program such as Miditzer or jOrgan (or other programs) which use Fluidsynth as their built in synthesizer there is no provision provided to with a single adjustment fine tune the pitch up or down to tune the synthesizer with another instrument. Is there a way to send a midi NRPN or RPN or other midi message to adjust the master tune of Fluidsynth?

**Comment by derselbst** [comment, marked as answer]
Fluidsynth [supports tuning for individual MIDI-channels as well as for banks / programs via RPN](https://github.com/FluidSynth/fluidsynth/wiki/FluidFeatures#rpn-control-change-implementation-chart). It also supports the SysEx commands of the MIDI Tuning Standard. I've never used any of them myself, so I can't really help you any further. Also, I don't know whether this would satisfy your idea of "master tuning". At least that kind of "master tuning" that you may know from e.g. Yamaha SysEx is not supported, AFAIK.

**Reply by mawe42** [reply]
As Tom said, FluidSynth supports the MIDI Tuning Standard. If you are unsure how to create the SysEx messages to adjust all notes up or down, you could use a tool like [Scala](http://www.huygens-fokker.org/scala/) to create the Tuning you need and export it as MTS messages.

If you are using MTS messages, please note that FluidSynth channels do not use a tuning bank/progam per channel by default. So contrary to the MIDI Tuning Standard, you can't simply change the tuning of tuning bank 0, tuning prog 0 to affect all channels. You will need to explicitly set all channels tuning bank and prog before your changes can be heard.

A simpler approach might be to simply use channel pitch bend messages. If Miditizer and/or jOrgan don't use pitch bend for real-time performance, then you could send one pitch bend message per channel after changing your channel sounds.

**Comment by jimhen3ry** [comment]
With the Miditzer, there is no provision for passing a user MIDI message to MIDI Out, which is what is sent on to FluidSynth by the Miditzer. What can be done is to route MIDI Out from the Miditzer to MIDI-OX using a virtual MIDI cable, such as loopMIDI. You can then merge the Miditzer's MIDI Out with your MIDI messages in MIDI-OX and send the merged MIDI to an external instance of FluidSynth, perhaps using QSynth. There are other ways of merging the Miditzer's MIDI Out with your MIDI messages and maybe another approach will be better than my off the top of my head, untested suggestion.
